lib/process-services-cloud/src/lib/people/services/people-cloud.service.spec.ts

```diff
@@ -0,0 +1,197 @@
+        findUserByEmailSpy = spyOn(identityUserService, 'findUserByEmail').and.returnValue(of([mockUsers[2]]));
+    }));
+
+    it('Should be able to call an API to search users based on search term', (done) => {
+        const searchTerm = 'searchTerm';
+        peopleCloudService.findUsers(searchTerm).subscribe((response) => {
+            expect(response.length).toBe(3);
+            expect(findUsersByNameSpy).toHaveBeenCalledWith(searchTerm);
+            done();
+        });
+    });
+
+    it('Should be able to fetch clientId based on application name', (done) => {
+        const mockAppName = 'mock-appname';
+        const mockClientId = 'mock-client-id';
+        peopleCloudService.getClientIdByApplicationName(mockAppName).subscribe((clientId) => {
+            expect(clientId).toBe(mockClientId);
+            expect(getClientIdByApplicationNameSpy).toHaveBeenCalledWith(mockAppName);
+            done();
+        });
+    });
+
+    it('Should thrown an error and not to call an API to fetch clientId if appName is not defined', (done) => {
+        peopleCloudService.getClientIdByApplicationName(null).subscribe(() => {},
+        (error) => {
+            expect(error).toBe('appName is mandatory to fetch clientId based on the appname');
+            expect(getClientIdByApplicationNameSpy).not.toHaveBeenCalled();
+            done();
+        });
+    });
+
+    it('Should be able to catch an error if API failes to fetch clientId based on appName', (done) => {
+        getClientIdByApplicationNameSpy.and.returnValue(throwError('Failed to fetch clientId'));
+        peopleCloudService.getClientIdByApplicationName('mock-appname').subscribe(() => {},
+        (error) => {
+            expect(error).toBe('Failed to fetch clientId');
+            expect(getClientIdByApplicationNameSpy).toHaveBeenCalled();
+            done();
+        });
+    });
+
+    it('Should be able to call an API on search results to check whether user has access to the application or not', (done) => {
+        const mockClientId = 'mock-client-id';
+        const searchTerm = 'searchTerm';
+        peopleCloudService.findUsersBasedOnApp(mockClientId, [], searchTerm).subscribe((response) => {
+            expect(response.length).toBe(3);
+            expect(checkUserHasClientAppSpy).toHaveBeenCalledTimes(3);
+            done();
+        });
+    });
+
+    it('Should be able to call an API on search results to check whether given roles mapped with client roles', (done) => {
+        const mockClientId = 'mock-client-id';
+        const mockRoles = ['MOCK_USER_ROLE'];
+        peopleCloudService.findUsersBasedOnApp(mockClientId, mockRoles, 'searchTerm').subscribe((response) => {
+            expect(response.length).toBe(3);
+            expect(checkUserHasAnyClientAppRoleSpy).toHaveBeenCalledTimes(3);
+            done();
+        });
+    });
+
+    it('Should not list users who does not have access to the applicatin', (done) => {
```

Comment:
You have a typo here. It should be `Should not list users who do not have access to the application`

Reply:
Updated